src/components/SearchTable.vue

```diff
@@ -28,18 +33,47 @@
         align: 'center'
       }))
     }
+
+    get filterOne () {
+      return this.headers.filter(h => h.isFilterable)[0]
+    }
+
+    get filterTwo () {
+      return this.headers.filter(h => h.isFilterable)[1]
+    }
+
+    get filterThree () {
+      return this.headers.filter(h => h.isFilterable)[2]
+    }
+
+    get filteredItems () {
+      let fItems = this.items
+      if (this.filterOne && this.filter[this.filterOne.value]) {
+        fItems = fItems.filter((i: any) => this.filter[this.filterOne.value] && this.filterOne.filterFunction(i, this.filter[this.filterOne.value]))
+      }
+      if (this.filterTwo && this.filter[this.filterTwo.value]) {
+        fItems = fItems.filter((i: any) => this.filter[this.filterTwo.value] && this.filterTwo.filterFunction(i, this.filter[this.filterTwo.value]))
+      }
+      if (this.filterThree && this.filter[this.filterThree.value]) {
+        fItems = fItems.filter((i: any) => this.filter[this.filterThree.value] && this.filterThree.filterFunction(i, this.filter[this.filterThree.value]))
+      }
+      return fItems
+    }
   }
 </script>
 
 <template lang="pug">
-  div(v-if=" items.length")
+  div(v-if="filteredItems")
     v-card
       v-card-title
-        v-spacer
-        v-spacer
-        v-spacer
         v-text-field(v-model="search", append-icon="fa-search", label="Search", single-line, hide-details)
-    v-data-table(:headers="alignedHeaders", v-bind="{ items, search }", :pagination.sync="pagination")
+        v-select(v-if="filterOne", :items="filterOne.filterChoices", :label="filterOne.text", v-model="filter[filterOne.value]", clearable, single-line, hide-details, :multiple="filterOne.isMultiSelect")
+        v-select(v-if="filterTwo", :items="filterTwo.filterChoices", :label="filterTwo.text", v-model="filter[filterTwo.value]", clearable, single-line, hide-details, :multiple="filterTwo.isMultiSelect")
+        v-select(v-if="filterThree", :items="filterThree.filterChoices", :label="filterThree.text", v-model="filter[filterThree.value]", clearable, single-line, hide-details, :multiple="filterThree.isMultiSelect")
+        v-spacer(v-if="!filterOne")
+        v-spacer(v-if="!filterTwo")
+        v-spacer(v-if="!filterThree")
```

Comment:
We should be able to handle more than three filters. Also, I added some margins to make the text more readable.
```
        v-text-field(v-model="search", append-icon="fa-search", label="Search", single-line, hide-details).ma-2
        v-select(
          v-for="header in headersWithFilters",
          :key="header.text",
          :items="header.filterChoices",
          :label="header.text",
          v-model="filterSelections[header.text]",
          clearable,
          single-line,
          hide-details,
          :multiple="header.isMultiSelect"
        ).ma-2
```